### PR TITLE
Migrate build air-gapped job 132

### PR DESCRIPTION
**Pull request by TrushynAliaksandr**

### CODE REVIEW

File: .buildkite/scripts/publish_air_gapped_docs.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+set +x
+DOCKER_USERNAME=$(vault kv get -field=username secret/ci/elastic-docs/docker.elastic.co)
+DOCKER_PASSWORD=$(vault kv get -field=password secret/ci/elastic-docs/docker.elastic.co)
+source .buildkite/scripts/retry.sh
+retry 5 docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD docker.elastic.co
+unset DOCKER_USERNAME
+unset DOCKER_PASSWORD
+set -x
+
+cd "$(git rev-parse --show-toplevel)"
```
Comment (nkammah) [inline .buildkite/scripts/publish_air_gapped_docs.sh:14]: is this needed?

File: air_gapped/build_bk.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+cd $(git rev-parse --show-toplevel)
+
+if [[ ! -d /opt/git-mirrors/built-docs.git ]]; then
+  echo "/opt/git-mirrors/built-docs.git must exist and contain a reference clone of the built-docs repo"
+  exit 1
+fi
+
+# Get an up to date copy of the repo
+rm -rf air_gapped/work
```
Comment (nkammah) [inline air_gapped/build_bk.sh:17]: is this needed if the agent is clean?

File: .buildkite/scripts/publish_air_gapped_docs.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+source ./air_gapped/build_bk.sh
+docker push "$AIR_GAPPED"
```
Comment (nkammah) [inline .buildkite/scripts/publish_air_gapped_docs.sh:16]: I know this was as in in the Jenkins world, but it's weird to push to $AIR_GAPPED where we don't see where it was sourced from. Maybe we should add AIR_GAPPED as either passed in or falling back on a default path? and same in the build script. If we actually do that, we can override the registry path in the top level job directly (in the pipeline.yml file) - that'll altogether be more readable and cleaner.

Comment (TrushynAliaksandr) [reply .buildkite/scripts/publish_air_gapped_docs.sh:16]: i can move it to pipeline yaml it is make sense, but i don't think that we need to override registry path using ui. Does it make sense? just put export to pipeline.yaml file

File: air_gapped/build_bk.sh (hunk 2 above)
Comment (TrushynAliaksandr) [reply air_gapped/build_bk.sh:17]: agree, I forgot that the agent is being cleaned

File: .buildkite/scripts/publish_air_gapped_docs.sh (hunk 3 above)
Comment (nkammah) [reply .buildkite/scripts/publish_air_gapped_docs.sh:15]: I was thinking of doing something like: 

```
AIR_GAPPED_REGISTRY_PATH="${AIR_GAPPED_REGISTRY_PATH:=docs-private/air_gapped}"
AIR_GAPPED=docker.elastic.co/$AIR_GAPPED_REGISTRY_PATH:latest 
```
in both the build.sh and publish.sh script, and simply adding the below in the pipeline.yaml

```
env:
   AIR_GAPPED_REGISTRY_PATH=docs-private/bk_air_gapped
```

Comment (TrushynAliaksandr) [reply .buildkite/scripts/publish_air_gapped_docs.sh:15]: hmmm, i don't know, looks too much for me, maybe do something like:

File: .buildkite/preview_cleaner_pipeline.yml [hunk 4]
```diff
@@ -1,3 +1,6 @@
 steps:
-  - label: "Example Test"
-    command: echo "Hello World!"
+  - label: ":white_check_mark: Preview cleaning"
+    command: bash .buildkite/scripts//clean_preview_branches.sh
```
Comment (nkammah) [inline .buildkite/preview_cleaner_pipeline.yml:3]:
```suggestion
    command: bash .buildkite/scripts/clean_preview_branches.sh
```

File: .buildkite/scripts/publish_air_gapped_docs.sh (hunk 3 above)
Comment (nkammah) [reply .buildkite/scripts/publish_air_gapped_docs.sh:15]: The intent is to allow folks to run the script locally without having it throw an error - but I guess we wouldn't want them to push to the production path anyway.

File: .buildkite/preview_cleaner_pipeline.yml (hunk 4 above)
Comment (TrushynAliaksandr) [reply .buildkite/preview_cleaner_pipeline.yml]: done

File: .buildkite/air_gapped_pipeline.yml [hunk 5]
```diff
@@ -1,3 +1,8 @@
 steps:
-  - label: "Example Test"
-    command: echo "Hello World!"
+  - label: ":white_check_mark: Build air-gapped"
+    command: |
```
Comment (nkammah) [inline .buildkite/air_gapped_pipeline.yml:3]: nit pick - but let's do the below

```
        command: ".buildkite/scripts/publish_air_gapped_docs.sh"
        env:
          AIR_GAPPED=docker.elastic.co/docs-private/bk_air_gapped:latest
```

File: .buildkite/scripts/clean_preview_branches.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -eo pipefail
+
+set +x
+GITHUB_TOKEN=$(vault read -field=value secret/ci/elastic-docs/docs_preview_cleaner)
```
Comment (nkammah) [inline .buildkite/scripts/clean_preview_branches.sh:6]:
```suggestion
export GITHUB_TOKEN=$(vault read -field=value secret/ci/elastic-docs/docs_preview_cleaner)
```

File: .buildkite/scripts/clean_preview_branches.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+export GITHUB_TOKEN
```
Comment (nkammah) [inline .buildkite/scripts/clean_preview_branches.sh:7]:
```suggestion
```

**Review by nkammah — APPROVED** [review]
added some minor nit-pick for cleanliness / readability - but lgtm

File: .buildkite/scripts/publish_air_gapped_docs.sh (hunk 3 above)
Comment (TrushynAliaksandr) [reply .buildkite/scripts/publish_air_gapped_docs.sh:17]: okay, let it be

**Pull request merged**